cluster-autoscaler/hack/update-mods.sh

```diff
@@ -0,0 +1,206 @@
+    k8s.io/kube-proxy
+    k8s.io/kube-scheduler
+    k8s.io/kubectl
+    k8s.io/kubelet
+    k8s.io/legacy-cloud-providers
+    k8s.io/metrics
+    k8s.io/node-api
+    k8s.io/sample-apiserver
+    k8s.io/sample-cli-plugin
+    k8s.io/sample-controller
+)
+
+################################################################################
+### Flag and directory validation ##############################################
+################################################################################
+
+if [[ $(basename $(pwd)) != "cluster-autoscaler" ]];then
+  echo "The script must be run in cluster-autoscaler directory"
+  exit 1
+fi
+AUTOSCALE_DIR=$(pwd)
+
+if ! which jq > /dev/null; then
+  echo "This script requires jq command to be available"
+  exit 1
+fi
+
+WORK_DIR="${WORK_DIR:-$(mktemp -d /tmp/ca-update-vendor.XXXX)}"
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h | --help ) PRINT_HELP="true"; break;;
+        -f | --k8sfork ) K8S_FORK="$2"; shift; shift ;;
+        -r | --k8srev ) K8S_REV="$2"; shift; shift ;;
+        -d | --workdir ) WORK_DIR="$2"; shift; shift ;;
+        -n | --no-update ) UPDATE_GO_MOD="false"; shift ;;
+        * ) break ;;
+    esac
+done
+
+if [[ "${PRINT_HELP}" == "true" ]]; then
+    cat << EOF >> /dev/stderr
+$SCRIPT_NAME - Updates or detects differences in go.mod, go.sum, /vendor
+
+    -f --k8sfork    The Kubernetes fork to use. Defaults to $K8S_FORK
+    -r --k8srev     Kuberntes revision to use. Defaults to ${K8S_REV}
+    -d --workdir    Working dir. Defaults to a temporary dir
+    -n --no-update  Set flag to only view differences in go.mod, /vendor
+    -h --help       Print this help output
+EOF
+    exit 0
+fi
+
+export GO111MODULE=on
+
+################################################################################
+### Ensure k8s.io/kubernetes version and overrides in mod file #################
+################################################################################
+if [ "$UPDATE_GO_MOD" == "true" ]; then
+    # Drop commit hash as version
+    go mod edit -require=k8s.io/kubernetes@$K8S_REV
+    # Get go toolchain to replace commit hash with pseudo version
+    go mod tidy
+    K8S_PSEUDO_VERSION=$(go mod edit -json | jq -r '.Require[] | select(.Path == "k8s.io/kubernetes") | .Version')
+
+    ## Add staging module overrides.
+    ## Extraneous replacements are harmless
+    for mod in ${K8S_STAGING_MODULES[*]}; do
+        go mod edit -require=${mod}@v0.0.0
+        go mod edit -replace=${mod}=k8s.io/kubernetes/staging/src/${mod}@${K8S_PSEUDO_VERSION}
+    done
+fi
+
+################################################################################
+### Get k8s.io/kubernetes/go.mod@$K8S_REV ######################################
+################################################################################
+cd $WORK_DIR
+K8S_GO_MOD_JSON="k8s-go-mod.json"
+curl -O $WORK_DIR/go.mod -s "https://raw.githubusercontent.com/${K8S_FORK}/${K8S_REV}/go.mod"
```

Comment:
This should be `-o`. With `-O` the `go.mod` file is not saved at all. Did this work for you?

Reply:
Derp, that was the last thing I added. I'll fix that